diff --git a/node_modules/@swc-node/register/lib/register.js b/node_modules/@swc-node/register/lib/register.js
index 7d4cfd8..750e1b2 100644
--- a/node_modules/@swc-node/register/lib/register.js
+++ b/node_modules/@swc-node/register/lib/register.js
@@ -88,7 +88,7 @@ function compile(sourcecode, filename, options, async = false) {
 exports.compile = compile;
 function register(options = {}, hookOpts = {}) {
     if (!process.env.SWCRC) {
-        options = (0, read_default_tsconfig_1.readDefaultTsConfig)();
+        options = Object.keys(options) ? options : (0, read_default_tsconfig_1.readDefaultTsConfig)();
     }
     options.module = ts.ModuleKind.CommonJS;
     (0, sourcemap_support_1.installSourceMapSupport)();
